app/src/main/java/org/thoughtcrime/securesms/conversation/newmessage/NewMessageFragment.kt

```diff
@@ -0,0 +1,182 @@
+            setDataAndType(requireActivity().intent.data, requireActivity().intent.type)
+            putExtra(ConversationActivityV2.THREAD_ID, DatabaseComponent.get(requireContext()).threadDatabase().getThreadIdIfExistsFor(recipient))
+        }.let(requireContext()::startActivity)
+        delegate.onDialogClosePressed()
+    }
+}
+
+@Preview
+@Composable
+private fun PreviewNewMessage(
+    @PreviewParameter(ThemeResPreviewParameterProvider::class) themeResId: Int
+) {
+    PreviewTheme(themeResId) {
+        NewMessage(State())
+    }
+}
+
+private val TITLES = listOf(R.string.enter_account_id, R.string.qrScan)
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun NewMessage(
+    state: State,
+    errors: Flow<String> = emptyFlow(),
+    callbacks: Callbacks = object: Callbacks {},
+    onClose: () -> Unit = {},
+    onBack: () -> Unit = {},
+    onHelp: () -> Unit = {},
+) {
+    val pagerState = rememberPagerState { TITLES.size }
+
+    Column(modifier = Modifier.background(MaterialTheme.colors.primarySurface)) {
+        AppBar(stringResource(R.string.messageNew), onClose = onClose, onBack = onBack)
+        SessionTabRow(pagerState, TITLES)
+        HorizontalPager(pagerState) {
+            when (TITLES[it]) {
+                R.string.enter_account_id -> EnterAccountId(state, callbacks, onHelp)
+                R.string.qrScan -> MaybeScanQrCode(errors, onScan = callbacks::onScanQrCode)
+            }
+        }
+    }
+}
+
+@Composable
+fun EnterAccountId(
+    state: State,
+    callbacks: Callbacks,
+    onHelp: () -> Unit = {}
+) {
+    Column(
+        modifier = Modifier.padding(horizontal = 12.dp, vertical = 16.dp).fillMaxHeight(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(16.dp)
+    ) {
+        SessionOutlinedTextField(
+            text = state.newMessageIdOrOns,
+            modifier = Modifier.padding(horizontal = 64.dp)
+                .contentDescription("Session id input box"),
+            placeholder = stringResource(R.string.accountIdOrOnsEnter),
+            onChange = callbacks::onChange,
+            onContinue = callbacks::onContinue,
+            error = state.error?.string(),
+        )
+        if (state.error == null) {
+            BorderlessButtonSecondary(
+                text = stringResource(R.string.messageNewDescription),
+                modifier = Modifier.contentDescription(R.string.AccessibilityId_help_desk_link)
+            ) { onHelp() }
+        }
+
+        OutlineButton(
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .padding(horizontal = 64.dp)
+                .width(200.dp)
+                .contentDescription(R.string.next),
+            onClick = { callbacks.onContinue() }
+        ) {
+            LoadingArcOr(state.loading) {
+                Text(stringResource(R.string.next))
```

Comment:
This text has no style?

Reply:
This has been changed back to a Text with no style...

Reply:
I don't want to repeat all of the fields and pass them to the text every time, and calculate the color based on disabled sttate:

```
Button( Slim, danger, enabled) {
    Text("hide", Slim, if(enabled) danger else LocalColors.disabled
}
```

It's simpler to inherit it:

```
Button(Slim, outline, danger, enabled) {
    Text("hide")
}
```

or with a courtesy:

```
SlimOutlineButton(danger, enabled) {
    Text("hide")
}
```

And the code is much more composable now, I can create any combination of size/style, color, outline/fill, enabled, loading content or temporaryClickContent, which was tedious and error-prone before.

_We could perhaps theoretically create a lint rule to make sure all `Text` has a style... unless its ultimately used in a `ButtonScope`..._

Reply:
Yes I like the modular approach of the buttons. I'm afraid for how it leaves the code at the declaration site though.
For example looking at the code above it's not clear at all what style the text will get. As a new developer looking at this you would need to dig in and finally find the `applyConstraints` code to understand that a style is inherited.